src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/EnderOffensiveSpell.java

```diff
@@ -0,0 +1,104 @@
+package com.mineboundteam.minebound.magic.OffensiveSpells;
+
+import com.mineboundteam.minebound.config.IConfig;
+import com.mineboundteam.minebound.item.armor.ArmorTier;
+import com.mineboundteam.minebound.magic.ActiveSpellItem;
+import com.mineboundteam.minebound.magic.MagicType;
+import com.mineboundteam.minebound.magic.SpellType;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class EnderOffensiveSpell extends ActiveSpellItem {
+    private final int manaCost;
+    private final int teleportDistance;
+
+    public EnderOffensiveSpell(Properties properties, EnderOffensiveSpellConfig config) {
+        super(properties, config.LEVEL, MagicType.ENDER, SpellType.OFFENSIVE);
+
+        this.manaCost = config.MANA_COST.get();
+        this.teleportDistance = config.TP_DISTANCE.get();
+    }
+
+    @Override
+    public void use(ItemStack stack, Level level, Player player) {
+        if (!level.isClientSide()) {
+            float direction = player.getYRot();
+            double x = (0 - Math.sin(direction * Math.PI / 180f));
+            double z = (Math.cos(direction * Math.PI / 180f));
+            int distance = 1;
+            while (distance <= teleportDistance) {
+                BlockPos tpLocation = new BlockPos(player.getPosition(1).add(distance * x, 0, distance * z));
```

Comment:
Shouldn't this work in 3 dimensions?

Reply:
It can if we want. I can pretty much take my math from the Fire offensive spell to make it work.

Reply:
Yeah, it feels restrictive currently

Reply:
Done. Used similar logic to what you used in Earth Defensive. Cleaned up the code quite a bit.